StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1119StatementMustNotUseUnnecessaryParenthesis.cs

```diff
@@ -60,7 +66,61 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(HandleParenthesizedExpressionn, SyntaxKind.ParenthesizedExpression);
+        }
+
+        private void HandleParenthesizedExpressionn(SyntaxNodeAnalysisContext context)
+        {
+            var node = context.Node as ParenthesizedExpressionSyntax;
+
+            if (node != null && node.Expression != null)
+            {
+                if (!(node.Expression is BinaryExpressionSyntax) && !(node.Expression is AssignmentExpressionSyntax)
```

Comment:
If you are going to separate this expression on the `&&` operator (which you should here due to length), can you go ahead and separate it on every `&&` operator at the same precedence within the expression?